src/lightning/pytorch/CHANGELOG.md

```diff
@@ -168,6 +168,11 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 - The `FSDPStrategy.load_optimizer_state_dict` and `FSDPStrategy.load_model_state_dict` are a no-op now ([#18358](https://github.com/Lightning-AI/lightning/pull/18358))
 
 
+- The `*_step(dataloader_iter)` flavor now no longer takes the `batch_idx` in the signature ([#18390](https://github.com/Lightning-AI/lightning/pull/18390))
+- Calling `next(dataloader_iter)` now returns a triplet `(batch, batch_idx, dataloader_idx)` ([#18390](https://github.com/Lightning-AI/lightning/pull/18390))
```

Comment:
TODO: check: does it always do that, even when using a single dataloader?

Reply:
Yes. The only alternative would be that it returns a dict-like object that has the attributes `batch`, `batch_idx`, `dataloader_idx`